core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/scatter3D/Scatter3dChartOpDesc.scala

```diff
@@ -0,0 +1,122 @@
+{
+  "attributeTypeRules": {
+    "title": "string"
+  }
+}
+""")
+class Scatter3dChartOpDesc extends VisualizationOperator with PythonOperatorDescriptor {
+  @JsonProperty(value = "title", required = true)
+  @JsonSchemaTitle("Title")
+  @JsonPropertyDescription("chart title")
+  var title: String = ""
+
+  @JsonProperty(value = "x", required = true)
+  @JsonSchemaTitle("Value X")
+  @JsonPropertyDescription("the value x axis")
+  @AutofillAttributeName
+  var x: String = ""
+
+  @JsonProperty(value = "y", required = true)
+  @JsonSchemaTitle("Value Y")
+  @JsonPropertyDescription("the value y axis")
+  @AutofillAttributeName
+  var y: String = ""
+
+  @JsonProperty(value = "z", required = true)
+  @JsonSchemaTitle("Value Z")
+  @JsonPropertyDescription("the value z axis")
+  @AutofillAttributeName
+  var z: String = ""
+
+  override def getOutputSchema(schemas: Array[Schema]): Schema = {
+    Schema.newBuilder.add(new Attribute("html-content", AttributeType.STRING)).build
+  }
+
+  override def operatorInfo: OperatorInfo =
+    OperatorInfo(
+      "Scatter3D",
+      "Visualize data in a Scatter3D Plot",
+      OperatorGroupConstants.VISUALIZATION_GROUP,
+      inputPorts = List(InputPort()),
+      outputPorts = List(OutputPort())
+    )
+
+  private def createPlotlyFigure(): String = {
+    assert(x.nonEmpty)
+    assert(y.nonEmpty)
+    assert(z.nonEmpty)
+    s"""
+       |        fig = go.Figure(data=[go.Scatter3d(
+       |            x=table["$x"],
+       |            y=table["$y"],
+       |            z=table["$z"],
+       |            mode='markers',
+       |            marker=dict(
+       |                size=12,
+       |                colorscale='Viridis',
+       |                opacity=0.8
+       |            )
+       |        )])
+       |        fig.update_traces(marker=dict(size=5, opacity=0.8))
+       |        fig.update_layout(
+       |            title='$title',
+       |            scene=dict(
+       |                xaxis_title='X: $x',
+       |                yaxis_title='Y: $y',
+       |                zaxis_title='Z: $z'
+       |            ),
+       |            margin=dict(t=40, b=30, l=10, r=10)
+       |        )
+       |""".stripMargin
+  }
+
+  override def generatePythonCode(): String = {
+    val finalcode =
+      s"""
+         |from pytexera import *
+         |
+         |import plotly.express as px
+         |import plotly.graph_objects as go
+         |import plotly.io
+         |import pandas as pd
+         |import numpy as np
+         |
+         |class ProcessTableOperator(UDFTableOperator):
+         |    def render_error(self, error_msg):
+         |        return '''<h1>Chart is not available.</h1>
+         |                  <p>Reason is: {} </p>
+         |               '''.format(error_msg)
+         |
+         |    @overrides
+         |    def process_table(self, table: Table, port: int) -> Iterator[Optional[TableLike]]:
+         |        original_table = table
```

Comment:
Is this variable useful? If not please remove it.

Reply:
Fixed.